file-echo-api/file-echo-api/Main.hs

```diff
@@ -6,24 +6,37 @@ module Main ( main ) where
 import qualified Data.Aeson as JSON
 import           Data.ByteString ( ByteString )
 import Data.Text (Text)
+import qualified Options.Applicative as Opt
 
 import qualified Argo as Argo
-import Argo.DefaultMain ( defaultMain )
+import Argo.DefaultMain ( customMain, userOptions )
 
 
 import qualified FileEchoServer as FES
 
 main :: IO ()
-main =
-  do theApp <- Argo.mkApp mkInitState serverMethods
-     defaultMain description theApp
+main = customMain parseServerOptions parseServerOptions parseServerOptions description getApp
+  where
+    getApp opts =
+      Argo.mkApp (mkInitState $ userOptions opts) serverMethods
 
 description :: String
 description =
   "An RPC server for loading and printing files."
 
-mkInitState :: (FilePath -> IO ByteString) -> IO FES.ServerState 
-mkInitState = const $ FES.initialState
+mkInitState :: ServerOptions -> (FilePath -> IO ByteString) -> IO FES.ServerState
+mkInitState opts reader = FES.initialState (initialFile opts) reader
+
+newtype ServerOptions = ServerOptions { initialFile :: Maybe FilePath }
+
+parseServerOptions :: Opt.Parser ServerOptions
```

Comment:
Might be nice to add a note here saying this function parses the _custom/additional_ options the server uses, and the handling of the standard options (e.g., http, stdio, socket) are still handled by the Argo backend.

Reply:
Done!